pkg/traceql/ast.go

```diff
@@ -1143,8 +1143,24 @@ func (a *MetricsAggregate) initFinal(q *tempopb.QueryRangeRequest) {
 	}
 }
 
-func (a *MetricsAggregate) observe(span Span) {
-	a.agg.Observe(span)
+func (a *MetricsAggregate) observe(span Span, traceID []byte) {
+	if len(traceID) == 0 {
+		a.agg.Observe(span)
+		return
+	}
+
+	var value float64
+	switch a.op {
+	case metricsAggregateQuantileOverTime,
```

Comment:
As discussed offline, this logic here that checks the operation and field needs to be moved to the init functions, so it's a "compile-time" step and not a "runtime-step".   Also it needs to be updated to look at the field being used in the aggregation (not necessarily duration), and duration needs to be scaled to seconds.  Other field types like int, need to be converted to float.  I'm not sure we can use the existing `bucketize..` functions since they are also chunking the value into the buckets.  I'd be ok with a little bit of duplication to have 2 new dedicated functions:  `getExemplarValueDuration` that returns/scales duration, and `getExemplarValue` that does a generic attribute lookup, and converts the field to float64 based on it's type.